src/d3d8/d3d8_device.cpp

```diff
@@ -935,15 +935,31 @@ namespace dxvk {
     // SetDepthStencilSurface is a separate call
     D3D8Surface* zStencil = static_cast<D3D8Surface*>(pNewZStencil);
 
-    if (likely(m_depthStencil != zStencil)) {
-      StateChange();
-      res = GetD3D9()->SetDepthStencilSurface(D3D8Surface::GetD3D9Nullable(zStencil));
+    // Depth stencil dimensions can not be lower than
+    // those of the currently set render target.
+    if (m_renderTarget != nullptr && zStencil != nullptr) {
```

Comment:
You're no longer checking whether `zStencil` is different from the one thats bound rn.

Reply:
Yeah, that's because stuff like changing the render target and setting the very same depth stencil can error out now depending on the result of the dimensions check. That check is done on d3d9 side anyway, so it will return early anyway in most cases.